facter/facter.php

```diff
@@ -141,8 +141,8 @@ function Update()
 	// Find HW type id
 	// 2011-08-31 <neil.scholten@gamigo.com>
 	// * adjust format to match default Dictionary Sets
-	if ($facter['is_virtual']=="false")
-	{	
+	if ($facter['is_virtual']=="false" && $_GET['update_hw_type'] != 'false')
```

Comment:
This will result in a PHP warning when the request does not have the parameter. Perhaps this should be a call to `isCheckSet()` or a similar inline test.